### DISCUSSION TITLE
Add quick commands to select tabs in tab switcher

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#23932] [Feature Requests]
<img width="748" alt="image" src="https://github.com/zed-industries/zed/assets/5475262/7af58565-c344-4b84-8a1f-654dce7de3e2">

e.g.

- `⌘-1` to select `Cargo.toml`
- `⌘-2` to select `quantized.rs`
- `⌘-3` to select `mistral.rs`
- ...

It would be helpful if one day we have the ability to hide the tab bars and don't know the index of the tab.

**Comment by alygin** [comment]
Some random thoughts

**...on the shortcut modifier key:**

`⌘` won't work as a modifier for these shortcuts. Tab switcher confirms selection on releasing of the initial modifiers, so when the user releases `ctrl` (that was used as a part of `ctrl-tab`), the switcher closes. Thus, only `ctrl-{?}` shortcuts will work.

**...on the numbering:**

Numbering should probably start with the third item in the list, i.e. `ctrl-1` should be assigned to `mistral.rs`, because:
- the first item is always the currently active tab, so selecting it is just an equivalent to closing the tab switcher without switching anywhere, and we already have a shortcut for it &mdash; `ctrl-esc`.
- the second item (the previously active tab) is selected by default when the tab switcher is deployed using `ctrl-tab`, and if we want to confirm its selection, we just need to release `ctrl`, there's no need for a shortcut.

Though when we deploy the tab switcher using `ctrl-shift-tab` and the last item is selected by default, the logic of assigning shortcuts to the list items should probably be different :)

**...on other possible use of these shortcuts:** 

IntelliJ products use `ctrl-1`, `ctrl-2`, ... and some other `ctrl-{?}` shortcuts in their Switcher windows (also accessible via `ctrl-tab`) to quickly switch to some other places, not only opened buffers:

<img width="404" alt="IntelliJ Switcher" src="https://github.com/zed-industries/zed/assets/2101250/85ab7f2e-dd8d-4ab2-a8c1-e5c6d86fbfc4">

Zed doesn't have so many places to switch to, but maybe later it will. Though we can use letters for other places if we decide to develop the tab switcher in this direction.

**...on displaying shortcuts:**

It'll be very handy to display these shortcuts in the tab switcher. Just to free the user from having to count items. Something like this:

<img width="427" alt="zed-tab-switcher-numbers" src="https://github.com/zed-industries/zed/assets/2101250/e98d948e-51e9-4970-9a3f-4d6395764767">

Or just numbers, like in the IntelliJ's switcher.

**Comment by maxdeviant** [comment]
This is an interesting idea. Going to move to a discussion to talk about it more.
